File: backend/src/main/java/pl/kielce/tu/villageSim/model/World.java
package pl.kielce.tu.villageSim.model;

import lombok.Getter;
import pl.kielce.tu.villageSim.service.aStar.PathNode;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class World {

    @Getter
    @SuppressWarnings("InstantiationOfUtilityClass")
    public static World INSTANCE = new World();

    /* Properties of world */

    public static Boolean IS_ACTIVE = false;

    public static Integer SIZE_WIDTH = 0;

    public static Integer SIZE_HEIGHT = 0;

    /* Resources */

    public static Integer WOOD = 0;

    public static Integer ROCK = 0;

    public static Integer FOOD = 0;

    public static Integer GOLD = 0;

    /* Paths */

    public static Map<Long, List<PathNode>> unitPaths = new HashMap<>();

}

File: frontend/src/app/model/world/world.ts
import {Unit} from "./unit";
import {Structure} from "./structure";
import {Building} from "./building";

export class World {

  sizeWidth: number = 96;

  sizeHeight: number = 32;

  units: Unit[] = [];

  buildings: Building[] = [];

  structures: Structure[] = [];

  wood: number = 0;

  rock: number = 0;

  food: number = 0;

  unitsAmount: number = 0;

  buildingsAmount: number = 0;
}

